cli/cmd/component_dev.go

```diff
@@ -0,0 +1,359 @@
+			}
+		}
+	}
+
+	// Missing tasks we can do for the developer
+	//
+	// 1) Change directory to Root path
+	//    > Command: 'cd ...'
+	// 2) Initialize git repository
+	//    > Command: 'git init'
+	// 3) Create your initial commit
+	//    > Command: 'git add .; git commit -m "feat: init component"'
+	// 4) Dowload Go dependencies
+	//    > Command: 'make go-vendor'
+	// 5) Build the component
+	//    > Command: 'make build'
+	// 6) Run the component via the Lacework CLI
+	//    > Command: 'lacework <component_name> placeholder'
+	//
+	cli.StartProgress("Initializing Git repository...")
+	err = cdkInitGitRepo(rootPath)
+	cli.StopProgress()
+	if err != nil {
+		cli.OutputChecklist(failureIcon, "Unable to initialize Git repository\n")
+		cli.Log.Debugw("unable to initialize Git repository", "error", err)
+	} else {
+		cli.OutputChecklist(successIcon, "Git repository initialized\n")
+	}
+
+	cli.StartProgress("Downloading Go dependencies...")
+	err = cdkGoVendor(rootPath)
+	cli.StopProgress()
+	if err != nil {
+		cli.OutputChecklist(failureIcon, "Unable to download Go dependencies\n")
+		cli.Log.Debugw("unable to download Go dependencies", "error", err)
+	} else {
+		cli.OutputChecklist(successIcon, "Go dependencies downloaded\n")
+	}
+
+	cli.StartProgress("Building your component...")
+	err = cdkGoBuild(rootPath)
+	cli.StopProgress()
+	if err != nil {
+		cli.OutputChecklist(failureIcon, "Unable to build your Go component\n")
+		cli.Log.Debugw("unable to build your Go component", "error", err)
+	} else {
+		cli.OutputChecklist(successIcon, "Dev component built at %s\n",
+			color.HiYellowString(filepath.Join(rootPath, component.Name)))
+	}
+
+	cli.StartProgress("Verifying component...")
+	err = cdkGoRunVerify(component.Name)
+	cli.StopProgress()
+	if err != nil {
+		// this is not on the developer, it's on this codebase, notify to fix it
+		cli.OutputChecklist(failureIcon, "Unable run scaffolding component\n")
+		cli.Log.Debugw("unable to run scaffolding component", "error", err)
+	} else {
+		cli.OutputChecklist(successIcon, "Component verified\n")
+	}
+
+	cli.OutputHuman("\nDeployment completed! Time for %s\n", randomEmoji())
+	return nil
+}
+
+func cdkInitGitRepo(rootPath string) error {
+	var (
+		vw  = terminal.NewVerboseWriter(10)
+		cmd = exec.Command("git", "init")
+	)
+	if _, err := vw.Write([]byte("\n")); err != nil {
+		cli.Log.Debugw("unable to write to virtual terminal", "error", err)
+	}
+	cmd.Env = os.Environ()
+	cmd.Dir = rootPath
+	cmd.Stdout = vw
+	cmd.Stderr = vw
+	if err := cmd.Run(); err != nil {
+		return errors.Wrap(err, "unable to initialize Git repo")
+	}
+
+	cmd = exec.Command("git", "add", ".")
+	cmd.Env = os.Environ()
+	cmd.Dir = rootPath
+	cmd.Stdout = vw
+	cmd.Stderr = vw
+	if err := cmd.Run(); err != nil {
+		return errors.Wrap(err, "unable to initialize Git repo")
+	}
+
+	cmd = exec.Command("git", "commit", "-m", "feat: init component")
```

Comment:
Should the running of git commands via exec.Command be replaced with a git package instead? 
https://github.com/go-git/go-git

Reply:
Very interesting, let me look at it 👀